fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -2831,4 +2856,47 @@ public static boolean isCompartmentSearch(FHIRSearchContext searchContext) {
         return result;
     }
 
+    /**
+     * Determine if a search parameter expression references an element of type Canonical.
+     * @param resourceType
+     * @param expression
+     * @return
+     */
+    private static boolean isCanonicalSearchParm(Class<?> resourceType, String expression) {
```

Comment:
given that this is called multiple times from multiple places, I'm a tad worried at the performance cost of this.

I *think* it would be possible to move this into some pre-processing and add an extension onto the SearchParameter definitions which can possibly target canonicals (similar to what we do for computing a token search params implicit system).  Thoughts?

Reply:
Thinking about this one some more, I'm not sure if the pre-processing idea is a good one or not.  In the case of implicit-systems, it works with or without the extension...having the extension just gives us a performance boost.

Since we allow users to configure their own search params, we probably don't want to rely on some precomputed extensions to get the proper system behavior.  The concern about the complexity/performance still stands though.  Maybe just a place to watch for as we move forward?

Reply:
I think that sounds reasonable. And this will only get called for Reference search parms, and only once per search parm, so hopefully not a big hit.

Reply:
OK, works for me.